pdns/dnsdistdist/doh.cc

```diff
@@ -567,9 +590,27 @@ static void processDOHQuery(DOHUnitUniquePtr&& unit)
       // but we should be fine as long as we don't touch du->req
       // outside of the main DoH thread
       du->status_code = 500;
-      sendDoHUnitToTheMainThread(std::move(du), "DoH killed in flight");
+      handleImmediateResponse(std::move(du), "DoH killed in flight");
       return;
     }
+
+    {
+      // if there was no EDNS, we add it with a large buffer size
+      // so we can use UDP to talk to the backend.
+      auto dh = const_cast<struct dnsheader*>(reinterpret_cast<const struct dnsheader*>(du->query.data()));
```

Comment:
Just to be sure, do we already know that `data()` is at least the size of a `dnsheader` at this point (otherwise we would not hit that point) ?

Reply:
Yes, otherwise it would have been discarded at the beginning of `doh_dispatch_query`.